### ISSUE TITLE
A datatype without constructor - what does it mean?

### ISSUE THREAD

**Issue by pjljvandelaar** [TorXakis/TorXakis#503]
TxsHappy.y defines
`Constructors : {- empty -} | Constructor | Constructors "|" Constructor`

Hence, one can define a data type without a constructor:
```
TYPEDEF MyType ::= ENDDEF
CHANDEF Channels ::=
    Stimulus :: MyType
ENDDEF
MODELDEF Spec ::=
    CHAN IN    Stimulus 
    CHAN OUT   

    BEHAVIOUR  
        Stimulus ? x
ENDDEF
```
Yet, what does it mean?

**Comment by pjljvandelaar** [comment]
When new datatypes are added, I propose to no longer accept it: ADTs must be constructable. Similar P ::= SingleConstructor(p :: P) should be rejected as a valid data type, since it can't be constructed.

**Comment by keremispirli** [comment]
Implemented in #565

Constructors with zero fields, on the other hand, are valid.
